### PR TITLE
Configure initial state for numlock/capslock

### PR DESCRIPTION
**Pull request by progandy**
Set numlock state to enabled by default, keep capslock disabled and make this configurable.

The initial state of numlock and capslock can be set with 
```input ... xkb_numlock|xkb_capslock disable|enable```

Implements #2311 

Note: This relies ```wlr_keyboard_notify_modifiers``` which is not meant to be called by the compositor, but I found no other option to reliably set the modifiers. 

I used the xkb_ prefix to indicate that setting the state will reload the keyboard configuration. 
The commands use enabled/disabled instead of off/on to match the other boolean input commands.

### CODE REVIEW

**Comment by SirCmpwn** [comment]
(still missing man page updates)

**Comment by progandy** [comment]
Now everything should be there. I had to clean up my local clone first.

File: sway/sway-input.5.scd [hunk 1]
```diff
@@ -33,6 +33,14 @@ For more information on these xkb configuration options, see
 *input* <identifier> xkb\_variant <variant>
 	Sets the variant of the keyboard like _dvorak_ or _colemak_.
 
+*input* <identifier> xkb\_capslock enabled|disabled
+	Enables or disables CapsLock on sway startup, the default is disabled.
+	Only meaningful if sway controls the keyboard.
+
+*input* <identifier> xkb\_numlock enabled|disabled
+	Enables or disables NumLock on sway startup, the default is enabled.
+	Only meaningful if sway controls the keyboard.
```
Comment (SirCmpwn) [inline sway/sway-input.5.scd]: This line can probably be removed from both.

**Comment by RedSoxFan** [comment]
Since the implementation is not restricted to startup, you could change the man page to `Enables or disables CapsLock/NumLock. The default on startup is disabled/enabled.`

**Comment by progandy** [comment]
I can do that. Should I somehow document that changing the value for NumLock resets CapsLock to the value stored in xkb_capslock as well? Maybe a general note that changing one xkb_ value applies all stored xkb_ values again.

**Comment by SirCmpwn** [comment]
Can't we get the current state of the modifiers and OR/AND them with only the bits we want to flip?

**Comment by progandy** [comment]
The flags are currently set in the keyboard configuration and then applied with ```apply_keyboard_config``` / ```sway_keyboard_configure```. There is no way to know which flag change was requested. I think I'd need a way to do different things when
* sway detects a new keyboard (startup or hotplug)
* swaymsg is sent for an existing keyboard

**Comment by SirCmpwn** [comment]
I see. Let's take the easy way out for now and just make it effective in the config file (not at runtime)?

**Comment by progandy** [comment]
Is there a way to do that?

I think I could create a status flag in the input config that the next call should only update the marked modifier, but that feels like dirty code. There shouldn't be status information in a configuration structure.

**Comment by SirCmpwn** [comment]
Just add it to the appropriate list of commands in commands.c.

**Comment by RedSoxFan** [comment]
I think the runtime portion of this can be added to the keyboard part of #1779

**Comment by SirCmpwn** [comment]
Aye.

**Comment by progandy** [comment]
Somehow I missed that list, it is so short.

Should this stay an individual setting for each keyboard then or simply a global option for all keyboards? I think I'd prefer the global option then.

**Comment by SirCmpwn** [comment]
I like it individual, if you want global do `input * xkb_numlock enable`

**Comment by progandy** [comment]
I can't add a subcommand of input to the config_commands list. I'll mark this as WIP and keep on thinking, maybe I'll get a good idea.

**Comment by RedSoxFan** [comment]
> I can't add a subcommand of input to the config_commands list. I'll mark this as WIP and keep on thinking, maybe I'll get a good idea.

See https://github.com/swaywm/sway/blob/master/sway/commands/bar.c#L38 and
https://github.com/swaywm/sway/blob/master/sway/commands/bar.c#L49

**Comment by progandy** [comment]
I had another idea. What do you think about allowing the swaymsg command and just storing the new value? It will be applied only after hotplugging the keyboard or changing its layout.

**Comment by SirCmpwn** [comment]
Do whatever you want, just get something mergable. This feature is not that complicated.

**Comment by progandy** [comment]
@SirCmpwn, @RedSoxFan, the pull request is ready for review. I kept it restricted to the configuration file for now. Sometimes I try to perfect and polish too much.

PS: I had to rebase it against the current master.

**Comment by RedSoxFan** [comment]
Thanks!

**Pull request merged**